tests/SpecTests/Operation.php

```diff
@@ -127,6 +128,17 @@ public static function fromCommandMonitoring(stdClass $operation)
         return $o;
     }
 
+    public static function fromConvenientTransactions(stdClass $operation)
+    {
+        $o = new self($operation);
+
+        if (isset($operation->collectionOptions)) {
+            $o->collectionOptions = (array) $operation->collectionOptions;
+        }
```

Comment:
Ah, I removed the error expectation as the operations for `withTransaction` sometimes are expected to error out without explicitly specifying an `error` field in the operation. I re-added both result and error expectations, but only create an `ErrorExpectation` instance if there is an `error` field on the operation. In other cases, we don't fail on an operation error, as it may very well be expected to test the `withTransactions` helper.